modules/build/src/test/scala/scala/build/tests/BuildOptionsTests.scala

```diff
@@ -28,6 +28,43 @@ class BuildOptionsTests extends munit.FunSuite {
     )
   }
 
+  test("-S 3.nightly option works") {
+    val options = BuildOptions(
+      scalaOptions = ScalaOptions(
+        scalaVersion = Some("3.nightly"),
+        scalaBinaryVersion = None,
+        supportedScalaVersionsUrl =
```

Comment:
It' seems, that you don't have to set `supportedScalaVersionsUrl`, because this test doesn't test this functionality.

```suggestion
        supportedScalaVersionsUrl = None
```